### ISSUE TITLE
Equation numbers disappears in HTML format output

### ISSUE THREAD

**Issue by henrywoo** [rstudio/rmarkdown#644]
Dear Rmarkdown author,

I can use Rmarkdown + latex to output equation number in PDF format like:

\begin{equation}
f(x;\mu,\sigma^2) = \frac{1}{\sigma\sqrt{2\pi}} e^{ -\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2 }\label{eq:equation1}
\end{equation}

and then refer it later.

But this doesn't work for HTML output. Can you please help out?

Thanks!

**Comment by henrywuxad** [comment]
In http://rmarkdown.rstudio.com/, it says one can write one rmarkdown file and convert to other formats. I have written a file with Rmd, and it can generate pdf file correctly. I expect that I can get a html format output too. Thanks.

**Comment by yihui** [comment]
I know what the problem is (I provided a workaround in another package which is not ready for you to use yet), but it may be a little difficult to explain. Anyway, to be able to refer to an equation in the HTML output, you need to configure MathJax: http://mathjax.readthedocs.org/en/latest/tex.html#automatic-equation-numbering This could be done via `includes` option in YAML (http://rmarkdown.rstudio.com/html_document_format.html). Then in theory you can refer to the equation using `\ref{eq:equation1}`, but Pandoc will strip off the LaTeX command for HTML output, i.e. `\ref{}` will disappear. To avoid that, you could do `\\ref{eq:equation1}`, but this won't work for LaTeX output.

I think you should file a feature request to Pandoc to preserve `\ref{}` for HTML output.

**Comment by henrywoo** [comment]
Thanks Yihui, but that is a bad news I cannot use your secret weapon package:-( Can we change \ref to \\ref into a temp file and then generate HTML output? Do you know what is the command to generate HTML output? I am thinking maybe i can write a script to do this.

**Comment by yihui** [comment]
I do not want to hack this in the rmarkdown package, and as I said, it will be great for Pandoc to support `\ref{}` for HTML output instead of us hacking at it. If you want to do it by yourself, you probably need to define a custom output format with a pre-processor to replace `\ref{}` with `\\ref{}` in the Markdown input file. See `?rmarkdown::output_format` for more info. This requires you to understand quite a bit about the rmarkdown internals.

**Comment by silvaac** [comment]
Thank you Yihui, I am also trying to set up eq # and I was wandering if there is an example of how to exactly set up the YAML include... do I put the mathjax script before the body? And if so do I need to save a file with this script.  Thank you.

**Comment by yihui** [comment]
@silvaac It is no longer secret: you can use `bookdown::html_document2`

- https://bookdown.org/yihui/bookdown/markdown-syntax.html
- https://bookdown.org/yihui/bookdown/a-single-document.html

**Comment by silvaac** [comment]
Thank you very much!





---- On Fri, 05 Aug 2016 15:18:13 -0700 Yihui Xie &lt;notifications@github.com&gt;wrote ---- 




@silvaac It is no longer secret: you can use bookdown::html_document2

https://bookdown.org/yihui/bookdown/markdown-syntax.html

https://bookdown.org/yihui/bookdown/a-single-document.html

**Comment by henrywoo** [comment]
Hi Yihui, I remember there is a equation reference example in the urls you posted. But now i could not find it. Did you remove it or move it to somewhere?

**Comment by yihui** [comment]
@henrywoo It has been moved to the next section: https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html

**Comment by henrywoo** [comment]
`bookdown` looks awesome. Do you have a plan to add its feature to rmarkdown? I wish I could use the features in normal rmarkdown.

**Comment by yihui** [comment]
@henrywoo No, there are no plans to back-port bookdown features directly into rmarkdown. Two reasons: 1) bookdown is relatively new and I need more users to test it so I can be sure some of my hacks are robust enough; 2) It is trivially easy to use these features in normal R Markdown documents: https://bookdown.org/yihui/bookdown/a-single-document.html
